apps/webservice/src/app/[workspaceSlug]/_components/target-filter/TargetConditionRender.tsx

```diff
@@ -0,0 +1,418 @@
+import type {
+  ComparisonCondition,
+  EqualCondition,
+  LikeCondition,
+  MetadataCondition,
+  NullCondition,
+  RegexCondition,
+  TargetCondition,
+} from "@ctrlplane/validators/targets";
+import { useState } from "react";
+import { useParams } from "next/navigation";
+import { capitalCase } from "change-case";
+import {
+  TbArrowUpCircle,
+  TbChevronDown,
+  TbCopy,
+  TbDots,
+  TbPlus,
+  TbRefresh,
+  TbTrash,
+  TbX,
+} from "react-icons/tb";
+
+import { cn } from "@ctrlplane/ui";
+import { Button } from "@ctrlplane/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuGroup,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@ctrlplane/ui/dropdown-menu";
+import { Input } from "@ctrlplane/ui/input";
+import { Popover, PopoverContent, PopoverTrigger } from "@ctrlplane/ui/popover";
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@ctrlplane/ui/select";
+import {
+  TargetFilterType,
+  TargetOperator,
+} from "@ctrlplane/validators/targets";
+
+import { api } from "~/trpc/react";
+import { useMatchSorter } from "~/utils/useMatchSorter";
+
+type TargetConditionRenderProps = {
+  condition: TargetCondition;
+  onChange: (condition: TargetCondition) => void;
+  onRemove?: () => void;
+  depth?: number;
+  className?: string;
+};
```

Comment:
```suggestion
type TargetConditionRenderProps<T extends  TargetCondition> = {
  condition: T;
  onChange: (condition: T) => void;
  onRemove?: () => void;
  depth?: number;
  className?: string;
};
```

Then you can remove all the casting in the components